src/MicrosoftTeams.ts

```diff
@@ -188,14 +362,23 @@ namespace microsoftTeams {
         // Undocumented field used to mock the window for unit tests
         currentWindow = this._window as Window || window;
 
-        // Listen for messages post to our window
-        let messageListener = (evt: MessageEvent) => processMessage(evt);
-        currentWindow.addEventListener("message", messageListener, false);
-
         // If we are in an iframe, our parent window is the one hosting us (i.e., window.parent); otherwise,
         // it's the window that opened us (i.e., window.opener)
         parentWindow = (currentWindow.parent !== currentWindow.self) ? currentWindow.parent : currentWindow.opener;
 
+        // Listen for messages post to our window
+        let messageListener = (evt: MessageEvent) => processMessage(evt);
+
+        // tslint:disable-next-line: no-any
+        if (!parentWindow && currentWindow["teamsNativeClient"]) {
+            // assign the parent
```

Comment:
Please add a more elaborate comment to clarify the usage of this for frame-less scenario.

Reply:
done